Sources/Rules.swift

```diff
@@ -8070,4 +8070,60 @@ public struct _FormatRules {
             }
         }
     }
+
+    public let spacesWithGuard = FormatRule(help: "Remove space between guard and add spaces after last guard.",
+                                            disabledByDefault: true)
+    { formatter in
+        func leaveOrSetLinebreaksInIndexes(_ indexes: Set<Int>, linebreaksCount: Int) {
+            var alreadyHasLinebreaksCount = 0
+            for index in indexes {
+                guard let token = formatter.token(at: index) else {
+                    return
+                }
+                if token.isLinebreak {
+                    if alreadyHasLinebreaksCount == linebreaksCount {
+                        formatter.removeToken(at: index)
+                    } else {
+                        alreadyHasLinebreaksCount += 1
+                    }
+                }
+            }
+            if alreadyHasLinebreaksCount != linebreaksCount,
+               let firstIndex = indexes.first
+            {
+                formatter.insert(.linebreak("\n", 0), at: firstIndex)
+            }
+        }
+
+        let guardKeyword = Token.keyword("guard")
+        formatter.forEach(guardKeyword) { guardIndex, _ in
```

Comment:
I think it's more common to just do:

```suggestion
        formatter.forEach(.keyword("guard")) { guardIndex, _ in
```